kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -542,6 +659,216 @@ namespace Kratos
+			ComputeExtrapolatedGeometryIntersections(rElement, rNumCutEdges, rCutEdgesRatioVector, rExtraGeomNormal, rCutExtraEdgesRatioVector);
+		} else if (TDim == 3) {
+			if (rNumCutEdges == 3) {
+				// if all three cut edges share one node, then the element is intersected and not incised
+				if (CheckIfCutEdgesShareNode(rElement, rCutEdgesRatioVector)) {
+					return;
+				}
+				ComputeExtrapolatedGeometryIntersections(rElement, rNumCutEdges, rCutEdgesRatioVector, rExtraGeomNormal, rCutExtraEdgesRatioVector);
+			} else if (rNumCutEdges == 1 || rNumCutEdges == 2) {
+				ComputeExtrapolatedGeometryIntersections(rElement, rNumCutEdges, rCutEdgesRatioVector, rExtraGeomNormal, rCutExtraEdgesRatioVector);
+			}
+		}
+	}
+
+	template<std::size_t TDim>
+	void CalculateDiscontinuousDistanceToSkinProcess<TDim>::ComputeExtrapolatedGeometryIntersections(
+        const Element& rElement,
+		unsigned int& rNumCutEdges,
+		array_1d<double, (TDim == 2) ? 3 : 6>& rCutEdgesRatioVector,
+		array_1d<double,3>& rExtraGeomNormal,
+		array_1d<double, (TDim == 2) ? 3 : 6>& rCutExtraEdgesRatioVector)
+	{
+		const auto r_edges_container = rElement.GetGeometry().GenerateEdges();
+		constexpr std::size_t n_edges = (TDim == 2) ? 3 : 6;
+
+		// Calculate average point of intersection points from rCutEdgesRatioVector for intersection plane definition
+		array_1d<double,3> avg_base_point = ZeroVector(3);
+		for (std::size_t i_edge = 0; i_edge < n_edges; i_edge++) {
+			// Calculate point coordinates and add to avg_point
+			if (rCutEdgesRatioVector[i_edge] >= 0.0) {
+				avg_base_point += ConvertEdgeRatioToIntersectionPoint(r_edges_container[i_edge], rCutEdgesRatioVector[i_edge]);
+			}
+		}
+		avg_base_point /= rNumCutEdges;
+
+		// Calculate intersections of each edge of element, which is not cut already, with the intersection plane
+		for (std::size_t i_edge = 0; i_edge < n_edges; i_edge++) {
+			if (rCutEdgesRatioVector[i_edge] == -1.0) {
+				array_1d<double,3> extra_int_pt;
+				const Element::NodeType& edge_point_0 = r_edges_container[i_edge][0];
+				const Element::NodeType& edge_point_1 = r_edges_container[i_edge][1];
+				int is_intersection = IntersectionUtilities::ComputePlaneEdgeIntersection(
+					avg_base_point, rExtraGeomNormal, edge_point_0.Coordinates(), edge_point_1.Coordinates(), extra_int_pt);
+
+				// Calculate intersection ratio of edge and save it
+				if (is_intersection == 1) {
+					rCutExtraEdgesRatioVector[i_edge] = ConvertIntersectionPointToEdgeRatio(r_edges_container[i_edge], extra_int_pt);
+				}
+			}
+		}
+	}
+
+	template<std::size_t TDim>
+	void CalculateDiscontinuousDistanceToSkinProcess<TDim>::ConvertEdgeDistancesToElementalDistances(
+		const Element& rElement,
+		const PointerVector<GeometricalObject>& rIntersectedObjects,
+		const array_1d<double, (TDim == 2) ? 3 : 6> &rCutEdgesRatioVector,
+        const array_1d<double, (TDim == 2) ? 3 : 6> &rCutExtraEdgesRatioVector,
+        Vector &rElementalDistancesExtraVector)
```

Comment:
Isn't the size of this array known at compile time? Shouldn't it be `TDim + 1`?

Reply:
Yes and I wanted to do it like that, but CorrectDistanceOrientation wants a Vector, I think because ELEMENTAL_DISTANCES are transferred directly, so I had to use Vector a couple of times, that's what it started with, if I remember correctly.

Reply:
OK. Makes sense. Thanks for clarifying.